### PR TITLE
fix(zone): make our typings comply with --noImplicitAny

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: zone/zone.d.ts [hunk 1]
```diff
@@ -48,8 +48,8 @@ declare class Zone {
 	static longStackTraceZone: {
 		getLongStacktrace(exception: any): string;
 		stackFramesFilter(line: string): boolean;
-		onError(exception): void;
-		fork(locals): Zone;
+		onError(exception: any): void;
+		fork(locals: any): Zone;
 	};
 	static getStacktrace(): Zone.Stacktrace;
 	static countingZone: {
```
Comment (vicb) [inline zone/zone.d.ts]: obsolete, should be removed

**Comment by vicb** [comment]
`exceptZone` should also be removed together with most of the static methods

File: zone/zone.d.ts [hunk 2]
```diff
@@ -9,23 +9,23 @@
...
-	bindOnce(fn): any;
-	run(fn, applyTo?, applyWith?): void;
+	bind(fn: Function, skipEnqueue: boolean): void;
+	bindOnce(fn: Function): any;
+	run(fn: Function, applyTo?: any, applyWith?: any): void;
 	beforeTask(): void;
 	onZoneCreated(): void;
 	afterTask(): void;
 	enqueueTask(): void;
 	dequeueTask(): void;
```
Comment (vicb) [inline zone/zone.d.ts]: missing methods:
- `isRootZone(): bool`
- `addEventListener`
- `removeEventListener` those 2 should have the same signature as their global counterpart

File: zone/zone.d.ts [hunk 3]
```diff
@@ -9,23 +9,23 @@
...
 
-	static patchSetClearFn(obj, fnNames): string;
```
Comment (vicb) [inline zone/zone.d.ts]: missing:
- `bindPromiseFn(fn: () => Promise<any>)` the fn can take any nb of args

**Comment by alexeagle** [comment]
Overwrote my patch with a version authored by @vicb

**Comment by alexeagle** [comment]
Merged as 31e7317c9a0793857109236ef7c7f223305a8aa9

**Pull request closed without merging**